wire-ios/Wire-iOS/Sources/UserInterface/UserProfile/Views/SecurityLevelView.swift

```diff
@@ -122,9 +109,52 @@ final class SecurityLevelView: UIView {
           iconImageView.widthAnchor.constraint(equalToConstant: 11.0),
           iconImageView.heightAnchor.constraint(equalToConstant: 11.0),
           iconImageView.centerYAnchor.constraint(equalTo: centerYAnchor),
-          iconImageView.trailingAnchor.constraint(equalTo: securityLevelLabel.leadingAnchor, constant: -10.0)
+          iconImageView.trailingAnchor.constraint(equalTo: securityLevelLabel.leadingAnchor, constant: -4.0)
         ])
     }
+
+    private func configureUpdatedCallingUI(with classification: SecurityClassification) {
+        switch classification {
+
+        case .classified:
+            securityLevelLabel.textColor = SemanticColors.Label.textSecurityEnabled
+            backgroundColor = SemanticColors.View.backgroundSecurityEnabled
+            iconImageView.image = Asset.Images.check.image.withTintColor(SemanticColors.Icon.backgroundSecurityEnabledCheckMark)
+            topBorder.backgroundColor = SemanticColors.View.borderSecurityEnabled
+
+        case .notClassified:
+            securityLevelLabel.textColor = SemanticColors.Label.textDefaultWhite
+            backgroundColor = SemanticColors.View.backgroundSecurityDisabled
+            iconImageView.image = Asset.Images.attention.image.withTintColor(SemanticColors.Icon.foregroundCheckMarkSelected)
+            topBorder.backgroundColor = .clear
+
+        default:
+            isHidden = true
+            assertionFailure("should not reach this point")
+        }
+    }
+
+    private func configureLegacyCallingUI(with classification: SecurityClassification) {
+        switch classification {
+
+        case .classified:
+            securityLevelLabel.textColor = SemanticColors.Label.textDefault
+            backgroundColor = SemanticColors.View.backgroundSecurityLevel
+            iconImageView.setIcon(.checkmark, size: .tiny, color: SemanticColors.Icon.backgroundCheckMarkSelected)
```

Comment:
I don't think that would work when switching from light to dark mode. Instead of `setIcon`. We need to use `setTemplateIcon`

```suggestion
            iconImageView.setTemplateIcon(.checkmark, size: .tiny)
            iconImageView.tintColor = SemanticColors.Icon.backgroundCheckMarkSelected
```

Reply:
Should we be changing this though? It's a legacy/deprecated UI. So my thinking was that I did not want to change logic on that legacy UI part.

Reply:
We should change it before actually, the previous implementation was not right. Somehow it slipped the cracks. That's the correct approach.

Reply:
👍